src/devices/BrickPi3/BrickPi3.cs

```diff
@@ -0,0 +1,948 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.BrickPi3.Models;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Device.Spi;
+using static Iot.Device.BrickPi3.SpiExceptions;
+using System.Device.Spi.Drivers;
+
+namespace Iot.Device.BrickPi3
+{
+    /// <summary>
+    /// The main Brick class allowing low level access to motors and sensors
+    /// </summary>
+    public class Brick
+    {
+        private const int ChipSelectLine = 1;
+        // To store the Sensor types as well as date to be sent when it's an I2C sensor
+        private SensorType[] _sensorType = { SensorType.None, SensorType.None, SensorType.None, SensorType.None };
+        private byte[] _i2cInBytes = { 0, 0, 0, 0 };
+        //Initernals to initalize the SPI
+        private static SpiDevice _brickPiSPI = null;
+        private BrickPiVoltage _brickPiVoltage = new BrickPiVoltage();
+
+        #region Properties
+
+        /// <summary>
+        /// used to store the SPI Address
+        /// used mainly when multiple bricks in a raw or not the default SPI address 
+        /// up to 254 addresses supported
+        /// </summary>
+        public byte SPIAddress { get; set; }
+
+        /// <summary>
+        /// Stores all the information regarding the Brick
+        /// This includes id, hardware and firmware versions
+        /// </summary>
+        public BrickPiInfo BrickPi3Info { get; internal set; }
+
+        /// <summary>
+        /// Stores the voltage information
+        /// </summary>
+        public BrickPiVoltage BrickPi3Voltage
+        {
+            get
+            {
+                _brickPiVoltage.Voltage3V3 = GetVoltage3V3();
+                _brickPiVoltage.Voltage5V = GetVoltage5V();
+                _brickPiVoltage.Voltage9V = GetVoltage9V();
+                _brickPiVoltage.VoltageBattery = GetVoltageBatteryVcc();
+                return _brickPiVoltage;
+            }
+        }
+
+        #endregion
+
+        #region Init and reset
+
+        /// <summary>
+        /// Initialize the brick including SPI communication
+        /// </summary>
+        /// <param name="spiAddress"></param>
+        public void InitSPI(byte spiAddress = 1)
```

Comment:
I don't know much of the specifics of this but would it make sense to move this to the constructor (and ResetAll to Dispose)?